keras_nlp/models/task.py

```diff
@@ -217,60 +226,129 @@ def from_preset(
                 "constructor with a `backbone` argument. "
                 f"Received: backbone={kwargs['backbone']}."
             )
-        preset_cls = check_config_class(preset)
-
-        # Backbone case.
-        if issubclass(preset_cls, Backbone):
-            if preset_cls is not cls.backbone_cls:
-                subclasses = list_subclasses(cls)
-                subclasses = tuple(
-                    filter(lambda x: x.backbone_cls == preset_cls, subclasses)
+
+        task = None
+        if check_file_exists(preset, TASK_CONFIG_FILE):
+            # Task case.
+            task_preset_cls = check_config_class(preset, TASK_CONFIG_FILE)
+            if not issubclass(task_preset_cls, cls):
+                raise ValueError(
+                    f"`{TASK_CONFIG_FILE}` has type `{task_preset_cls.__name__}` "
+                    f"which is not a subclass of calling class `{cls.__name__}`. Call "
+                    f"`from_preset` directly on `{task_preset_cls.__name__}` instead."
                 )
-                if len(subclasses) == 0:
-                    raise ValueError(
-                        f"No registered subclass of `{cls.__name__}` can load "
-                        f"a `{preset_cls.__name__}`."
-                    )
-                if len(subclasses) > 1:
-                    names = ", ".join(f"`{x.__name__}`" for x in subclasses)
-                    raise ValueError(
-                        f"Ambiguous call to `{cls.__name__}.from_preset()`. "
-                        f"Found multiple possible subclasses {names}. "
-                        "Please call `from_preset` on a subclass directly."
+            task = load_serialized_object(preset, TASK_CONFIG_FILE)
+            if load_weights:
+                task.load_task_weights(get_file(preset, TASK_WEIGHTS_FILE))
+                task.backbone.load_weights(get_file(preset, MODEL_WEIGHTS_FILE))
```

Comment:
load tokenizers assets here, and then early return. preprocessor is already loaded

Reply:
preprocessor seems to be still None at this point! I did this for now:
```
task.preprocessor = task.preprocessor_cls.from_preset(preset)
```
but I agree that the following should create the preprocessor object:
```
task = load_serialized_object(preset, TASK_CONFIG_FILE)
```
I'll debug to see why preprocessor is None here!

Reply:
It was a test model issue! Done!